src/Core/NBB.Core.Effects/DependencyInjectionExtensions.cs

```diff
@@ -12,6 +12,8 @@ public static IServiceCollection AddEffects(this IServiceCollection services)
             services.AddSingleton(typeof(Thunk.Handler<>));
             services.AddSingleton(typeof(Parallel.Handler<,>));
             services.AddSingleton(typeof(Sequenced.Handler<>));
+            services.AddSingleton(typeof(TryWith.Handler<>));
+            services.AddSingleton(typeof(TryFinally.Handler<>));

```

Comment:
We should register all side effect handlers that depend on the interpreter as scoped